wordpress-mobile/WordPress-iOS
# New Time Zone Suggester

Timezone selection from site settings in iOS app is outdated in comparison with WordPress web and Android.  This PR adds unit tests to the ViewModel and a small refactor for better readability

Ref [11450](https://github.com/wordpress-mobile/WordPress-iOS/issues/11450)

To test:

1. Navigate to My Site -> Site Settings -> Time Zone
2. Make sure the time zone that is suggested is consistent with the timezone of the device / simulator you are testing on
3. Check that there's a GMT offset string below the TimeZone primary title and the local time for that TimeZone is shown as a subtitle on the right hand side
4. Run TimeZoneSelectorViewModelTests and Tests should succeed

<img width="237" alt="image" src="https://user-images.githubusercontent.com/88816724/147210865-ee71a193-2af5-4bba-8d69-4742b15ea918.png">


## Regression Notes
1. Potential unintended areas of impact
TimeZone selector screen was changed - unintended impact could be user would not be able to change their TimeZone

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I retested all steps in PR Add new TimeZoneSuggester [#17145](https://github.com/wordpress-mobile/WordPress-iOS/pull/17145) and [#17214](https://github.com/wordpress-mobile/WordPress-iOS/pull/17214)

- Navigate to My Site -> Site Settings -> Time Zone
- Make sure the time zone that is suggested is consistent with the time zone of the device / simulator you are testing on
- Tap the suggested time zone
- This will navigate you back to Site Settings screen where you should see the Suggested time you have chosen
- Tap Time Zone again from Site Settings screen
- To test search results works normally and has no regression:
- Search for a time zone (example Casablanca). TableView results should filter and show Casablanca.
- Tap Casablanca
- This will navigate you back to Site Settings screen where you should see the Suggested time you have chosen (Casablanca)

3. What automated tests I added (or what prevented me from doing so)
I isolated ViewModel logic in its own class and created UnitTests to test the ViewModel

PR submission checklist:
